### ISSUE TITLE
No selectable effect

### ISSUE THREAD

**Issue by glvercellone** [Schneegans/Burn-My-Windows#118]
## Describe the Bug
I can't set any effects on the setup screen. I am always offered an empty-menu!
(See the screen-shot)

1. Go to Welcome window (First window of the extension: Burn my window setting)
2. Click on --> Chose of animation Menu-list
4. See error --> no choice is allowed, the menu is empty!

You may also check the output of GNOME Shell for any error messages.
This can be done with the following terminal command.
Execute this in a terminal and observe the output while reproducing the bug.

```bash
journalctl -f -o cat | grep -E 'burn-my-windows|' :

(EE) event18 - Corsair Corsair STRAFE Gaming Keyboard: client bug: event processing lagging behind by 27ms, your system is too slow
Started Application launched by gnome-shell.
[session uid=1000 pid=73172] Activating via systemd: service name='org.gnome.Terminal' unit='gnome-terminal-server.service' requested by ':1.176' (uid=1000 pid=95359 comm="/usr/bin/gnome-terminal -- fish")
Starting GNOME Terminal Server...
[session uid=1000 pid=73172] Successfully activated service 'org.gnome.Terminal'
Started GNOME Terminal Server.
Started VTE child process 95399 launched by gnome-terminal-server process 95364.
[2022-03-27 12:20:03.617] [info] cat(95412)
[2022-03-27 12:20:03.619] [info] rm(95422)
[2022-03-27 12:20:03.674] [info] pacman(95584)
```

## System
 - Linux Garuda 
 - Burn-My-Windows version 12
 - GNOME Shell version: 42.0-0.1
 -
![Screenshot from 2022-03-27 12-13-38](https://user-images.githubusercontent.com/54698397/160276791-dc3dd4e2-839e-4df9-8552-fc8e4f4f97c5.png)

**Comment by Schneegans** [comment]
Thanks for the report! There was a bug in the version check for GNOME 42. (it used to work for `42.alpha` and `42.beta` but not for `42.0`). I just uploaded a new version to extensions.gnome.org which should hopefully fix the issue! If not, please leave a comment here :wink:

**Comment by glvercellone** [comment]
Tested and all is it OK.
Perfect, great Job!
Ciao.

**Comment by MrTomRod** [comment]
I experience the same bug! My system:

- Fedora 36 beta
- Gnome 42
- Wayland
- Burn-My-Windows  13

```text
$ journalctl -f -o cat | grep -E 'burn-my-windows|'
JS ERROR: TypeError: window.get_titlebar().pack_end is not a function
PreferencesDialog/<@/home/thomas/.local/share/gnome-shell/extensions/burn-my-windows@schneegans.github.com/prefs.js:228:33
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:150:31
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:186:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:21:13
run@resource:///org/gnome/gjs/modules/script/package.js:206:19
start@resource:///org/gnome/gjs/modules/script/package.js:190:8
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
systemd-hostnamed.service: Deactivated successfully.
SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
BPF prog-id=0 op=UNLOAD
BPF prog-id=0 op=UNLOAD
Object 0x55ce48d1b940 of type IBusText has been finalized while it was still owned by gjs, this is due to invalid memory management.
Object 0x55ce45b8a0a0 of type IBusText has been finalized while it was still owned by gjs, this is due to invalid memory management.
Object 0x55ce46445180 of type IBusText has been finalized while it was still owned by gjs, this is due to invalid memory management.
Object 0x55ce46445180 of type IBusText has been finalized while it was still owned by gjs, this is due to invalid memory management.
Window manager warning: Ping serial 3128684 was reused for window W15, previous use was for window W90.
dbus-:1.2-org.gnome.Extensions@1.service: Consumed 5.315s CPU time.
Object 0x7f0bc8a02e60 of type IBusText has been finalized while it was still owned by gjs, this is due to invalid memory management.
```

**Comment by Schneegans** [comment]
That's weird :confused: Have you restarted your session after upgrading to Burn-My-Windows 13?

If yes, what's the output of `gnome-shell --version`?

**Comment by MrTomRod** [comment]
Yes, I reinstalled and rebooted to confirm the bug persists.

```bash
$ gnome-shell --version
GNOME Shell 42.rc
```

**Comment by Schneegans** [comment]
Ahh, that explains it, I didn't know about `rc`. I just thought there is `alpha` and `beta`. So I guess it's time for another release :D. I'll fix this later today! Thanks for the report!

**Comment by MrTomRod** [comment]
Thanks a lot for your incredible work and quick responses!

But why does it break so quickly?! Does the `gnome-shell` API change every day? 😅

**Comment by Schneegans** [comment]
Well no, it does not change that quickly, but a few things changed between `42.alpha` and `42.beta`. So there are some conditional branches in the code depending on this version string. Up to now, I just checked for those two (and all upcoming minor versions such as `42.0` and `42.1`). However, there seems to be a `42.rc` between `42.beta` and `42.0` which I did not handle. And therefore BMW behaved like `42.alpha` on `42.rc` which caused the error :smile:
